lib/jerboa/client/credentials.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Jerboa.Client.Credentials do
+  @moduledoc false
+  ## Data structure containing client's authentication data
+
+  defstruct [:username, :realm, :secret, :nonce]
+
+  @type t :: %__MODULE__{
```

Comment:
I'd be keen on seeing this refactored in the light of this idea: https://www.youtube.com/watch?v=kZ1P8cHN3pY&feature=youtu.be&t=21m27s
@arkgil what do you think about trying to do so? I'm not sure how many state changes we do before all fields are  defined, so it might not be worth the effort in this particular case. I'm leaving it to your judgment.